File: Communication entre processus/exo1.py
import os
import sys

def main():
    
    r, w = os.pipe()

    pid = os.fork()

    if pid == 0:
        
        mot = input("Saisir un mot : ")
        
      
        data_to_send = mot.ljust(50).encode()
        os.write(w, data_to_send)
        
        
        res_bytes = os.read(r, 4)
        resultat = int.from_bytes(res_bytes, byteorder=sys.byteorder)
        
        print(f"Résultat : {resultat}")
        os._exit(0)

    else:
     
        data_received = os.read(r, 50).decode().strip()
        
        trouve = 0
        try:
            with open("texte.txt", "r") as f:
                contenu = f.read().split()
                if data_received in contenu:
                    trouve = 1
        except FileNotFoundError:
            print("Erreur : Le fichier 'texte.txt' n'existe pas.")

       
        os.write(w, trouve.to_bytes(4, byteorder=sys.byteorder))
        
        
        os.wait()

if __name__ == "__main__":
    main()

File: Communication entre processus/exo1.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

int main() 
{
    int p[2];
    pipe(p);

    if (fork() == 0) 
    {
        
        char mot[50];
        int resultat;
        printf("Saisir un mot : ");
        scanf("%s", mot);

        write(p[1], mot, 50);     
        read(p[0], &resultat, sizeof(int)); 
        
        printf("Résultat : %d\n", resultat);
        exit(0);
    } 
    else 
    {
        
        char mot_cherche[50], mot_fichier[50];
        int trouve = 0;
        FILE *f = fopen("texte.txt", "r");

        read(p[0], mot_cherche, 50); 

        if (f) 
        {
            while (fscanf(f, "%s", mot_fichier) != EOF) 
            {
                if (strcmp(mot_cherche, mot_fichier) == 0) 
                {
                    trouve = 1;
                    break;
                }
            }
            fclose(f);
        }

        write(p[1], &trouve, sizeof(int)); 
    }
    return 0;
}